extensions-contrib/kafka-emitter/src/test/java/org/apache/druid/emitter/kafka/KafkaEmitterTest.java

```diff
@@ -19,95 +19,459 @@
+      new AlertEvent("service1", "host1", "description"),
+      new AlertEvent("service2", "host2", "description")
+  );
+
+  private static final List<Event> REQUEST_LOG_EVENTS = ImmutableList.of(
+      DefaultRequestLogEventBuilderFactory.instance().createRequestLogEventBuilder(
+          "requests",
+          RequestLogLine.forSql(
+              "sql1", null, DateTimes.nowUtc(), null, new QueryStats(ImmutableMap.of())
+          )
+      ).build("service", "host"),
+      DefaultRequestLogEventBuilderFactory.instance().createRequestLogEventBuilder(
+          "requests",
+          RequestLogLine.forSql(
+              "sql2", null, DateTimes.nowUtc(), null, new QueryStats(ImmutableMap.of())
+          )
+      ).build("service", "host")
+  );
+
+  private static final List<Event> SEGMENT_METADATA_EVENTS = ImmutableList.of(
+      new SegmentMetadataEvent(
+          "ds1",
+          DateTimes.of("2001-01-01T00:00:00.000Z"),
+          DateTimes.of("2001-01-02T00:00:00.000Z"),
+          DateTimes.of("2001-01-03T00:00:00.000Z"),
+          "ds1",
+          true
+      ),
+      new SegmentMetadataEvent(
+          "ds2",
+          DateTimes.of("2020-01-01T00:00:00.000Z"),
+          DateTimes.of("2020-01-02T00:00:00.000Z"),
+          DateTimes.of("2020-01-03T00:00:00.000Z"),
+          "ds2",
+          true
+      )
+  );
+
+  private static final List<Event> UNKNOWN_EVENTS = ImmutableList.of(
+      new TestEvent(),
+      new TestEvent(),
+      new TestEvent()
+  );
+
+  @Test(timeout = 10_000)
+  public void testServiceMetricEvents() throws JsonProcessingException, InterruptedException
+  {
+    final KafkaEmitterConfig kafkaEmitterConfig = new KafkaEmitterConfig(
+        "",
+        new HashSet<>(Collections.singletonList(KafkaEmitterConfig.EventType.METRICS)),
+        "metrics",
+        "alerts",
+        "requests",
+        "segments",
+        "clusterName",
+        ImmutableMap.of("clusterId", "cluster-101"),
+        null,
+        null
+    );
+
+    final KafkaEmitter kafkaEmitter = initKafkaEmitter(kafkaEmitterConfig);
+
+    final List<Event> inputEvents = flattenEvents(SERVICE_METRIC_EVENTS);
+    final CountDownLatch eventLatch = new CountDownLatch(inputEvents.size());
+
+    final Map<String, List<String>> feedToExpectedEvents = trackExpectedEventsPerFeed(
+        inputEvents,
+        kafkaEmitterConfig.getClusterName(),
+        kafkaEmitterConfig.getExtraDimensions()
+    );
+    final Map<String, List<String>> feedToActualEvents = trackActualEventsPerFeed(eventLatch);
+
+    emitEvents(kafkaEmitter, inputEvents, eventLatch);
+
+    validateEvents(feedToExpectedEvents, feedToActualEvents);
+
+    Assert.assertEquals(0, kafkaEmitter.getMetricLostCount());
+    Assert.assertEquals(0, kafkaEmitter.getAlertLostCount());
+    Assert.assertEquals(0, kafkaEmitter.getSegmentMetadataLostCount());
+    Assert.assertEquals(0, kafkaEmitter.getSegmentMetadataLostCount());
```

Comment:
@abhishekrb19 typo error here? we have a double check on the same value

Reply:
Yes, good catch. Thanks for fixing it up in https://github.com/apache/druid/pull/15890